### ISSUE TITLE
[REST API Renderer] Wizard Cancel force redirect Response

### ISSUE THREAD

**Issue by thoresuenert** [laravel-arcanist/arcanist#19]
First of all we pushed our Renderer to GitHub today:
[suenerds/arcanist-rest-api-renderer](https://github.com/suenerds/arcanist-rest-api-renderer)

We tried to implement a cancel button today and found a `hardcoded` redirect response in the `Wizard destroy` method.

**Idea for a solution:**
You have build a delete hook for clean up purposes.
As we need a way to overwrite the redirect response I want to use the same mechanism for completing the wizard
```php
/**
     * The action that gets executed after the last step of the
     * wizard is completed.
     */
    protected string $onDeleteAction = NullAction::class;


/**
     * Gets called after the last step in the wizard is finished.
     */
    protected function onAfterDelete(ActionResult $result): Response | Responsable | Renderable
    {
        return new JsonResponse([
            'redirect' => [
                'path' => '/wizard/assignments'
            ]
        ]);
    }

 public function destroy(Request $request, string $wizardId): Response | Responsable | Renderable
    {
        $this->load($wizardId);

        $result = $this->actionResolver
            ->resolveAction($this->onDeleteAction)
            ->execute($this->transformWizardData());

        if (!$result->successful()) {
            return $this->responseRenderer->redirectWithError(
                $this->currentStep(),
                $this,
                $result->error()
            );
        }

        $this->wizardRepository->deleteWizard($this);

        return $this->onAfterDelete($result);
    }
```

The Delete Hook will be an `Action Class` after that and we can use  the `onAfterDelete` function to do whatever we need to do after Delete.

I will update this issue when my PR is ready.

**Comment by ksassnowski** [comment]
This is really two different issues:

1. Perform some sort of cleanup when a wizard gets deleted
2. Use the `ResponseRenderer` to return the response instead of hardcoding `redirect()`

Regarding the first issue, the `destroy` method already calls a `beforeDelete` hook that you can override to perform whatever cleanup necessary. By default, this is simply an empty method. (https://github.com/laravel-arcanist/arcanist/blob/main/src/AbstractWizard.php#L232). Adding another action class would add a lot of unnecessary overhead in this case in my opinion.

I definitely agree, however, that the `destroy` method should use the `ResponseRenderer`. That was an oversight on my part.

**Comment by thoresuenert** [comment]
> Regarding the first issue, the `destroy` method already calls a `beforeDelete` hook that you can override to perform whatever cleanup necessary. By default, this is simply an empty method. (https://github.com/laravel-arcanist/arcanist/blob/main/src/AbstractWizard.php#L232). Adding another action class would add a lot of unnecessary overhead in this case in my opinion.

Thats a good point. The `beforeDelete` hook should be flexible enough.
But I think in most cases you wouldn't touch the default `NullAction`.
It was just an idea to have the same implementation for both `very similar` cases.

> I definitely agree, however, that the `destroy` method should use the `ResponseRenderer`. That was an oversight on my part.

Moving a `redirect` method into the `ResponseRenderer` will be a bit tricky because the contract already have a redirect method. That would be a more cleaner way to tackle that problem.

What do you think about moving the `redirect()->` into a function `onAfterDelete`?

```php
/**
* Gets called after the last step in the wizard is finished.
*/
protected function onAfterDelete(): Response | Responsable | Renderable
{
    return redirect()->to($this->redirectTo());
}
```
with changing this Line: https://github.com/laravel-arcanist/arcanist/blob/9dc3fedbfa196704ac05da681f7983b4e898103f/src/AbstractWizard.php#L236
```php
return $this->onAfterDelete();
```
That does the job and isn't a breaking change for now.

Maybe we can rethink the Renderer to move all `routing` into one Class.

**Comment by ksassnowski** [comment]
Solved by #20
